src/main/java/use_cases/set_availability/SetAvailabilityUseCase.java

```diff
@@ -0,0 +1,55 @@
+package use_cases.set_availability;
+import entity_factories.*;
+import entity_layer.*;
+
+import java.time.LocalDateTime;
+import java.util.Arrays;
+import java.util.function.Consumer;
+
+/**
+ * Class responsible for creating a TimeManager entity with multiple TimeBlock items, created with the
+ * passed information
+ *
+ * @author ad-obas1187
+ */
+public class SetAvailabilityUseCase implements SetAvailabilityInputBoundary {
+
+    SetAvailabilityOutputBoundary availabilityPresenter;
+
+    public SetAvailabilityUseCase(SetAvailabilityOutputBoundary availabilityPresenter) {
+        this.availabilityPresenter = availabilityPresenter;
+    }
+
+    /**
+     * 1. Take in (String[]) list of availabilities inputted by User
+     * 2. Using that, create a CommonTimeBlockManager factory that further creates a CommonTimeBlockManager
+     * 3. Create a SetAvailability Presenter
+     * 4. Return that presenter
+     *
+     * @param availabilityInputs an array...
+     */
+    @Override
+    public void create(String[] availabilityInputs, int curriculumId) {
+
+        //Update TimeBlockManager to have availabilityInputs as LocalDate times
+        Schedule schedule = InMemoryUser.getActiveUser().getSchedule();
+        TimeBlockManager timeBlockManager = schedule.getAvailability();
+
+        for (String s : availabilityInputs) {
+            int len = s.length();
+            LocalDateTime start = LocalDateTime.parse(s.substring(0, len/2));
+            LocalDateTime end = LocalDateTime.parse(s.substring(len/2, len));
+
+            TimeBlockFactory timeBlockFactory = new CommonTimeBlockFactory();
```

Comment:
Overall, this looks very good! However, having the TimeBlockFactory be inside of your for-loop does two things: one, it creates unnecessary duplicates of a class that will be used identically at each iteration, and two, it hard couples your interactor class with the CommonTimeBlockFactory class. I recommend instead adding a new final attribute for the factory and setting it in the constructor, and you can call it within your for loop.